packages/flutter/test/widgets/scrollbar_test.dart

```diff
@@ -3635,4 +3635,79 @@ The provided ScrollController cannot be shared by multiple ScrollView widgets.''
     expect(size.height - rtlRect1.bottom, 4);
     expect(rtlRect1.left, 3);
   });
+
+  testWidgets(
+    'RawScrollbar does not assert when dynamically assigning controller with thumbVisibility true',
+    (WidgetTester tester) async {
+      // This test reproduces the issue from https://github.com/flutter/flutter/issues/172614
+      // where dynamically assigning a controller and setting thumbVisibility to true
+      // in the same frame caused an assertion failure.
+
+      final ScrollController scrollController = ScrollController();
+      addTearDown(scrollController.dispose);
+
+      bool useController = false;
+      bool thumbVisible = false;
+      bool interactive = false;
+
+      await tester.pumpWidget(
+        StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Directionality(
+              textDirection: TextDirection.ltr,
+              child: MediaQuery(
+                data: const MediaQueryData(),
+                child: Column(
+                  children: <Widget>[
+                    ElevatedButton(
+                      onPressed: () {
+                        setState(() {
+                          // This reproduces the exact scenario from the issue:
```

Comment:
nit: maybe this can be:

```
// This reproduces the exact scenario from the linked issue:
// Initially the controller is null, thumbVisibility is false, and interactive is false.
// Then, in the next frame, the controller is non-null, thumbVisibility is true, and interactive is true.
```